include/ck/tensor_operation/gpu/device/device_base.hpp

```diff
@@ -5,17 +5,47 @@
 
 #include <string>
 #include <sstream>
+#include <regex>
+#include <optional>
 
 #include "ck/stream_config.hpp"
 
 namespace ck {
 namespace tensor_operation {
 namespace device {
 
+#define GET_OBJECT_NAME_IMLP                                                \
+    std::optional<std::string> GetObjectName() const override               \
+    {                                                                       \
+        std::string Str = __PRETTY_FUNCTION__;                              \
+        static std::regex ObjNameExpr{"<std::string> (.*)::GetObjectName"}; \
+        std::smatch Match;                                                  \
+        if(!std::regex_search(Str, Match, ObjNameExpr))                     \
+        {                                                                   \
+            return Str;                                                     \
+        }                                                                   \
+        return std::string(Match[1]) + ';';                                 \
+    }
+
+#define GET_TEMPLATE_INFO_IMPL                                  \
+    std::optional<std::string> GetTemplateInfo() const override \
+    {                                                           \
+        std::string Str = __PRETTY_FUNCTION__;                  \
+        static std::regex TemplateExpr{"\\[(.*)\\]"};           \
+        std::smatch Match;                                      \
+        if(!std::regex_search(Str, Match, TemplateExpr))        \
+        {                                                       \
+            return std::nullopt;                                \
+        }                                                       \
+        return std::string(Match[1]);                           \
+    }
+
+#define REGISTER_EXTRA_PRINTING_METHODS GET_OBJECT_NAME_IMLP GET_TEMPLATE_INFO_IMPL
```

Comment:
Since you use few additional std headers I wonder whether just registering those functions won't lead to compiler errors in translation units where those headers won't be included...

Reply:
Extra headers are defined in `device_base.hpp`. Every class which needs to be inherited from `BaseOperator` requires to include `device_base.hpp`. This will be done explicitly or implicitly - i.e., through a chain of `#include`s. It seems to me that everything should be fine.